examples/2d/text2d.rs

```diff
@@ -8,32 +8,65 @@ fn main() {
         .run();
 }
 
+struct AnimateTranslation;
+struct AnimateRotation;
+struct AnimateScale;
+
 fn setup(mut commands: Commands, asset_server: Res<AssetServer>) {
+    let font = asset_server.load("fonts/FiraSans-Bold.ttf");
+    let text_style = TextStyle {
+        font,
+        font_size: 60.0,
+        color: Color::WHITE,
+    };
+    let text_alignment = TextAlignment {
+        vertical: VerticalAlign::Center,
+        horizontal: HorizontalAlign::Center,
+    };
     // 2d camera
     commands.spawn_bundle(OrthographicCameraBundle::new_2d());
-    commands.spawn_bundle(Text2dBundle {
-        text: Text::with_section(
-            "This text is in the 2D scene.",
-            TextStyle {
-                font: asset_server.load("fonts/FiraSans-Bold.ttf"),
-                font_size: 60.0,
-                color: Color::WHITE,
-            },
-            TextAlignment {
-                vertical: VerticalAlign::Center,
-                horizontal: HorizontalAlign::Center,
-            },
-        ),
-        ..Default::default()
-    });
+    // Demonstrate changing translation
+    commands
+        .spawn_bundle(Text2dBundle {
+            text: Text::with_section("translation", text_style.clone(), text_alignment),
+            ..Default::default()
+        })
+        .insert(AnimateTranslation);
+    // Demonstrate changing rotation
+    commands
+        .spawn_bundle(Text2dBundle {
+            text: Text::with_section("rotation", text_style.clone(), text_alignment),
+            ..Default::default()
+        })
+        .insert(AnimateRotation);
+    // Demonstrate changing scale
+    commands
+        .spawn_bundle(Text2dBundle {
+            text: Text::with_section("scale", text_style, text_alignment),
+            ..Default::default()
+        })
+        .insert(AnimateScale);
 }
 
-fn animate(time: Res<Time>, mut query: Query<&mut Transform, With<Text>>) {
-    // `Transform.translation` will determine the location of the text.
-    // `Transform.scale` and `Transform.rotation` do not yet affect text (though you can set the
-    // size of the text via `Text.style.font_size`)
-    for mut transform in query.iter_mut() {
-        transform.translation.x = 100.0 * time.seconds_since_startup().sin() as f32;
+fn animate(
+    time: Res<Time>,
+    mut queries: QuerySet<(
+        Query<&mut Transform, WithBundle<(Text, AnimateTranslation)>>,
```

Comment:
I think these Queries should use the `Query<&mut Transform, (With<Text>, With<AnimateTranslation>)>` style instead. WithBundle isn't a pattern I want to encourage generally (and the behavior of it might change in the future to be "spawned with this specific bundle" instead of "has the components in this bundle"). For now, I want examples to focus on individual components.

Reply:
Ah! I had first (unsuccessfully) tried `With<(Text, AnimateTranslation)>` -- now I know the correct approach, I can use it!